### PR TITLE
Integrating Sym with Core: Network Architecture Layers- Part 2

### PR DESCRIPTION
**Pull request by mnabian**
# Modulus Pull Request

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/modulus/blob/main/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [x] The [CHANGELOG.md](https://github.com/NVIDIA/modulus/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/modulus/issues) is linked to this pull request.

### CODE REVIEW

File: modulus/models/interpolation.py [hunk 1]
```diff
@@ -0,0 +1,520 @@
...
+import enum
+import math
+from typing import Tuple, List
+
+import numpy as np
+import torch
+import torch.nn.functional as F
+from torch import Tensor
+
+# TODO enum causes segmentation faults with current torch script. Go back to enum after torch script update
```
[the remark this answers is not in the document]
Comment (mnabian) [reply modulus/models/interpolation.py:24]: I think it is a good idea to keep this until someone takes a closer look at this.

File: modulus/models/pix2pix/pix2pix.py [hunk 2]
```diff
@@ -106,7 +106,7 @@ class Pix2Pix(Module):
         Number of upsampling blocks, by default 3
     n_blocks : int, optional
         Number of residual blocks in middle of model, by default 3
-    activation_fn : str, optional
+    activation_fn : Any, optional
```
[the remark this answers is not in the document]
Comment (mnabian) [reply modulus/models/pix2pix/pix2pix.py:109]: It is a little bit tricky. Sym doesn't use the entry points, so there shouldn't be an issue there. But yes, I think we should not use the type `Any` here. @loliverhennigh any suggestions?

File: modulus/models/interpolation.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (mnabian) [reply modulus/models/interpolation.py]: Moved to the `layers' folder

File: modulus/models/pix2pix/pix2pix.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (mnabian) [reply modulus/models/pix2pix/pix2pix.py:113]: Changed to `str`.

[the remark this answers is not in the document]
Comment (mnabian) [reply modulus/models/pix2pix/pix2pix.py:113]: Opened an issue here to find a way to fix the type: https://github.com/NVIDIA/modulus/issues/185

**Review by loliverhennigh — APPROVED** [review]

**Review by ktangsali — APPROVED** [review]
Thanks for addressing the comments. LGTM.

**Pull request merged**